scenario/go/scenario_test.go

```diff
@@ -38,10 +40,14 @@ const (
 const defaultListenerName = ""
 
 var (
-	gateways                 map[string]*GatewayConnection
-	currentGateway           *GatewayConnection
-	transaction              *Transaction
-	lastCommittedBlockNumber uint64
+	gateways                        map[string]*GatewayConnection
+	currentGateway                  *GatewayConnection
+	transaction                     *Transaction
+	lastCommittedBlockNumber        uint64
+	checkpointer                    *client.InMemoryCheckpointer
+	lastBlockEventReceived          *common.Block
+	lastFilteredBlockEventReceived  *peer.FilteredBlock
+	lastBlockAndPrivateDataReceived *peer.BlockAndPrivateData
```

Comment:
`checkpointer` is stored on the GatewayConnection and the last block received are not used at all now that checkpointing is taken care of in the listener. These should all be removed.